components/script/layout_wrapper.rs

```diff
@@ -471,9 +471,13 @@ impl<'le> TElement for ServoLayoutElement<'le> {
         self.get_attr(namespace, attr).map_or(false, |x| x == val)
     }
 
-    fn insert_flags(&self, flags: ElementFlags) {
+    fn insert_selector_flags(&self, flags: ElementFlags) {
         self.element.insert_atomic_flags(flags);
     }
+
+    fn clear_selector_flags(&self) {
```

Comment:
Also, is there any way we can use the type system to guarantee that we only clear the flags during exclusive access? Would &mut self work here?